File: HakerRank/data_structures/arrays/2d_array.py
# author: Daniel Lozano
# source: HAckerRank ( https://www.hackerrank.com )
# problem name: Data Structures: Arrays: 2D Array - DS
# problem url: https://www.hackerrank.com/challenges/2d-array/problem
# date: 8/16/2017

array = []
for i in range(6):
    array.append(input().split())
    for j in range(6):
        array[i][j] = int(array[i][j])

higher = -999999999
for i in range(4):
    for j in range(4):
        add = array[i][j] + array[i][j+1] + array[i][j+2] + array[i+1][j+1] + array[i+2][j] + array[i+2][j+1] + array[i+2][j+2]
        if add > higher:
            higher = add

print(higher)

File: HakerRank/data_structures/arrays/2d_array.cs
/*
author: Daniel Lozano
source: HackerRank ( https://www.hackerrank.com )
problem name: Data Structures: Arrays: 2D Array - DS
problem url: https://www.hackerrank.com/challenges/2d-array/problem
*/

using System;

namespace HackerRank
{
    public class Resolution
    {
        public static void Main()
        {
            int[,] matrix = new int[6, 6];
            for (int i = 0; i < 6; i++)
            {
                int[] temp_matrix = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
                for (int j = 0; j < 6; j++)
                {
                    matrix[i, j] = temp_matrix[j];
                }
            }
            Console.Write(Solution(matrix));

        }
        public static int Solution(int[,] lines)
        {
            int sum = -9999999;
            for (int i = 0; i < 4; i++)
            {
                for (int j = 0; j < 4; j++)
                {
                    int currentSum = lines[i, j] + lines[i, j + 1] + lines[i, j + 2] + lines[i + 1, j + 1] + lines[i + 2, j] + lines[i + 2, j + 1] + lines[i + 2, j + 2];
                    if (currentSum > sum)
                        sum = currentSum;
                }
            }
            return sum;
        }
        
    }
}
